src/v2/dry_run.rs

```diff
@@ -0,0 +1,1250 @@
+use super::{
+    generated::{
+        self, account_transaction_payload, dry_run_error_response, dry_run_request,
+        DryRunInvokeInstance, DryRunMintToAccount, DryRunSignature, DryRunStateOperation,
+    },
+    AccountIdentifier, IntoBlockIdentifier,
+};
+use crate::{
+    types::{
+        smart_contracts::{ContractContext, InstanceInfo, ReturnValue},
+        AccountInfo, AccountTransactionDetails, RejectReason,
+    },
+    v2::{generated::DryRunStateQuery, Require},
+};
+use concordium_base::{
+    base::{Energy, ProtocolVersion},
+    common::types::{CredentialIndex, KeyIndex, Timestamp},
+    contracts_common::{AccountAddress, Amount, ContractAddress},
+    hashes::BlockHash,
+    smart_contracts::ContractTraceElement,
+    transactions::{EncodedPayload, PayloadLike},
+};
+use futures::*;
+
+mod shared_receiver {
+    use futures::{stream::Stream, StreamExt};
+    use tokio::{
+        sync::{mpsc, oneshot},
+        task::JoinHandle,
+    };
+
+    /// A `SharedReceiver` wraps an underlying stream so that multiple clients
+    /// can queue to receive items from the stream.
+    pub struct SharedReceiver<S>
+    where
+        S: Stream, {
```

Comment:
It usually adds only inconvenience to have a constraint on the type.

There are cases where it adds value, but I don't see this value here. The `impl` blocks constrain this further anyhow.